dm/pkg/checker/binlog.go

```diff
@@ -172,3 +170,81 @@ func (pc *MySQLBinlogRowImageChecker) Check(ctx context.Context) *Result {
 func (pc *MySQLBinlogRowImageChecker) Name() string {
 	return "mysql_binlog_row_image"
 }
+
+// BinlogDBChecker checks if migrated dbs are in binlog_do_db or binlog_ignore_db.
+type BinlogDBChecker struct {
+	db            *sql.DB
+	dbinfo        *dbutil.DBConfig
+	schemas       map[string]struct{}
+	caseSensitive bool
+}
+
+// NewBinlogDBChecker returns a RealChecker.
+func NewBinlogDBChecker(db *sql.DB, dbinfo *dbutil.DBConfig, schemas map[string]struct{}, caseSensitive bool) RealChecker {
+	newSchemas := make(map[string]struct{}, len(schemas))
+	for schema := range schemas {
+		newSchemas[schema] = struct{}{}
+	}
+	return &BinlogDBChecker{db: db, dbinfo: dbinfo, schemas: newSchemas, caseSensitive: caseSensitive}
+}
+
+// Check implements the RealChecker interface.
+func (c *BinlogDBChecker) Check(ctx context.Context) *Result {
+	result := &Result{
+		Name:  c.Name(),
+		Desc:  "check whether migrated dbs are in binlog_do_db/binlog_ignore_db",
+		State: StateFailure,
+		Extra: fmt.Sprintf("address of db instance - %s:%d", c.dbinfo.Host, c.dbinfo.Port),
+	}
+
+	flavor, err := utils.GetFlavor(ctx, c.db)
+	if err != nil {
+		markCheckError(result, err)
+		return result
+	}
+	binlogDoDB, binlogIgnoreDB, err := utils.GetBinlogDB(ctx, c.db, flavor)
+	if err != nil {
+		markCheckError(result, err)
+		return result
+	}
+	if !c.caseSensitive {
+		binlogDoDB = strings.ToLower(binlogDoDB)
+		binlogIgnoreDB = strings.ToLower(binlogIgnoreDB)
+	}
+	binlogDoDB = strings.ReplaceAll(binlogDoDB, " ", "")
```

Comment:
identifier can have spaces as long as they are backquoted, so this is incorrect

Reply:
In fact, there is no spaces among the list of `binlog_do_db`. So, I just remove this later.

But another question, it seems that identifier also can have `,`(comma)?